src/app/routes/features/features.component.ts

```diff
@@ -15,24 +16,68 @@ import {FeatureRendererComponent} from './feature-renderer.component';
 export class FeaturesComponent implements OnInit {
 
   features: Feature[];
+  filter: string;
   gridApi: GridApi;
   columnApi: ColumnApi;
+  pageSize = 2;
   gridOptions: GridOptions;
 
-  constructor(private featureService: FeatureService, private logger: NGXLogger) {
+  getQuickFilter = (params) => params.value;
+
+  constructor(private featureService: FeatureService,
+              private logger: NGXLogger,
+              private paginatorService: PaginatorService) {
+      const colDefs: ColDef[] = [
+          {
+              field: 'uid',
+              getQuickFilterText: this.getQuickFilter
+          },
+          {
+              field: 'description',
+              getQuickFilterText: this.getQuickFilter
+          },
+          {
+              field: 'group',
+              getQuickFilterText: this.getQuickFilter
+          },
+          {
+              field: 'permissions',
+              getQuickFilterText: this.getQuickFilter
+          }
+      ];
       this.gridOptions = {
           headerHeight: 0,
-          columnDefs: [],
+          columnDefs: colDefs,
           rowHeight: 160,
           suppressHorizontalScroll: true,
+          enableFilter: true,
           fullWidthCellRenderer: 'fullWidthCellRenderer',
           frameworkComponents: {
               fullWidthCellRenderer: FeatureRendererComponent
           },
-          isFullWidthCell: (rowNode: RowNode) => true
+          suppressPaginationPanel: true,
+          isFullWidthCell: (rowNode: RowNode) => true,
+          onGridReady: (params) => {
+              this.gridApi = params.api;
+              this.columnApi = params.columnApi;
+          },
+          pagination: true,
+          paginationPageSize: this.pageSize,
```

Comment:
Instead of `paginationPageSize: this.pageSize` use `paginationAutoPageSize: true,`